### PR TITLE
Implement a system clock and use for ipv4 and arp

### PR DESCRIPTION
**Pull request by sirkrypt0**
This PR introduces a system timer that updates the clock every second

### CODE REVIEW

File: kernel/network/routing.h [hunk 1]
```diff
@@ -21,14 +21,19 @@
 #pragma once
 
 #include "kernel/network/ethernet.h"
+#include "kernel/time.h"
 
 #define ROUTING_DEBUG
```
Comment (felixauringer) [inline kernel/network/routing.h]: We should comment out this as well

File: kernel/time.c [hunk 2]
```diff
@@ -0,0 +1,105 @@
...
+ *    GNU General Public License for more details.                            *
+ *                                                                            *
+ *    You should have received a copy of the GNU General Public License       *
+ *    along with this program.  If not, see <http://www.gnu.org/licenses/>.   *
+ ******************************************************************************/
+
+#include "time.h"
+#include "kernel/memory.h"
+#include <sys/types.h>
+#include <stdio.h>
```
Comment (felixauringer) [inline kernel/time.c]: Where do we need this?

**Comment by sirkrypt0** [comment]
We should create a user mode library for this and use a system call to get the clock time

https://filippo.io/linux-syscall-table/

**Review by felixauringer — CHANGES_REQUESTED** [review]
Looks very good!

**Comment by felixauringer** [comment]
> We should create a user mode library for this and use a system call to get the clock time
> 
> https://filippo.io/linux-syscall-table/

Can we please do this after the remaining exams?

**Review by felixauringer — APPROVED** [review]

**Pull request merged**